src/main/java/core/basesyntax/service/impl/TransactionParserImpl.java

```diff
@@ -0,0 +1,41 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.Operation;
+import core.basesyntax.service.TransactionParser;
+import java.util.ArrayList;
+import java.util.List;
+
+public class TransactionParserImpl implements TransactionParser {
+    @Override
+    public List<FruitTransaction> parse(List<String> transactionStrings) {
+        if (transactionStrings == null || transactionStrings.isEmpty()) {
+            throw new RuntimeException("Transactions parsing failed: "
+                    + "no transactions found");
+        }
+        List<FruitTransaction> transactions = new ArrayList<>();
+        String[] parts;
+        Operation operation;
+        String fruitName;
+        int quantity;
+        for (String line : transactionStrings) {
+            parts = line.split(",");
+            if (parts.length != 3) {
+                throw new RuntimeException("Transactions parsing failed: "
+                        + "wrong format in line: " + line);
```

Comment:
The exception message 'wrong format in line' could include the expected format to help the user understand what is correct.